### ISSUE TITLE
App Suggestion: [Appsmith]

### ISSUE THREAD

**Issue by fr3fou** [caprover/one-click-apps#280]
**Please tell us what app you'd like to see on CapRover as a one-click app.**

I'd like to see https://github.com/appsmithorg/appsmith

**Do you know if there is any official Docker image for the app?**

`appsmith/appsmith-server` and `appsmith/appsmith-editor`

**Comment by JosXa** [comment]
The #324 version is indeed a little over my head. No chance of configuring that without consulting the docs.

**Comment by aqidd** [comment]
This is what I did to setup #324 

1. Install Appsmith locally to get the initial database
2. Move the database somewhere (mongo export import operation)
3. Run #324 and point the database to number 2

**Comment by davidsparkles** [comment]
@aqidd thank you very much for your work on the appsmith-only one-click app. Till there is a one-click app for the complete appsmith stack, would you be so kind as to provide a more detailed step-by-step instruction on how to get the mongodb initialized locally, imported to the manually created mongodb in caprover and configuring the appsmith-server to point to that mongodb properly.

I spend two days without any success. I am stuck with the appsmith-server not beeing able to connect to the mongodb, that is also deployed on caprover.

**Comment by Jeyakumarrathnaselvi** [comment]
I am facing similar issues. can someone guide

**Comment by githubsaturn** [comment]
Looks like this app is lacking proper instructions and multiple people are having issues with it. I'll remove it in the next few days from the repository if no fix is provided.

**Comment by aqidd** [comment]
Running appsmith locally should be covered in their documentation here https://docs.appsmith.com/setup/docker

Should I update this with the link to documentation?

**Comment by githubsaturn** [comment]
@aqidd  - the docs that you linked cover the mongodb issues that were mentioned above? If yes, let's update to include the link.

**Comment by aqidd** [comment]
The migration from local mongodb to other mongodb instance should be handled like normal database migration. You just need to make sure that the encryption variables are the same.

I'm not sure if we should put too many details here since other apps doesn't as well. For example, wordpress/hasura/ghost installation without database does not cover how you deploy locally, setup the database and migrate the data.

**Comment by githubsaturn** [comment]
Makes sense. So if the one click app is working as expected, we can keep it. 

Optionally, if you want to write a tutorial on how to do the migration, we can link it in the description.

**Comment by aqidd** [comment]
Can confirm it is still working since I'm doing updates by creating new appsmith instance (with newer version) then moving the domain name from the older instance to the newer instance. 

I'll try to make the tutorial for this and update the app with description.

**Comment by timconsidine** [comment]
> Should I update this with the link to documentation?
It would be a lot more helpful to add a database.
Dozens of caprover apps have DB connections setup within their install.
And it is after all supposed to be a 1-click install

**Comment by timconsidine** [comment]
> For example, wordpress/hasura/ghost installation without database does not cover how you deploy locally, setup the database and migrate the data.

I bet you that no-one or very few bothers installing these apps.
If an app needs a DB, then it should be provided within the caprover install.

**Comment by timconsidine** [comment]
> Looks like this app is lacking proper instructions and multiple people are having issues with it. I'll remove it in the next few days from the repository if no fix is provided.

Yes, I think it should be removed as it is useless without one of EITHER (a) decent documentation (more than just "go add a database") OR (b) a DB connection embedded within the caprover install (like dozens of other app packagers have done).  

The real issue here seems to be that the app packager for Appsmith-NoDB doesn't know how to package a DB connection in the app, so hides behind excuses why there is not one.  First the app says NoDB and then forces you you to put a Mongo DB connection, when in fact Appsmith supports many different connections.   

This package is not helpful to the community in its current state and should be removed unless proper documentation or a DB connection is added.

**Comment by aqidd** [comment]
I will go with whatever @githubsaturn decide on this one. 
I created several packages with DB and no DB for caprover, however for this appsmith, my main use case is without database so I just create one without. If what's needed is 2 packages (with/without DB) then feel free to remove the current definition. I'll make them when I have time.

Cheers

**Comment by aqidd** [comment]
I added PR for appsmith with (embedded) mongodb database here https://github.com/caprover/one-click-apps/pull/594 . This is their official way of running with docker/docker-compose (https://docs.appsmith.com/setup/docker). I gave a quick check and looks fine. Kindly review @githubsaturn

**Comment by timconsidine** [comment]
Thank you so much !
An alternative Appsmith one-click has been added !  Yay ! 🍾 🙏 
Just installed and will play with it.
Very much appreciated, thank you.

**Comment by mredodos** [comment]
my appsmith cant start, need some env or config ?  @timconsidine  @aqidd

**Comment by timconsidine** [comment]
Sorry to hear that.
And sorry to report that I have stopped using Caprover.
Nice system but too many broken app templates, other self-host PaaS more stable.
